src/lib/bool.ts

```diff
@@ -311,37 +310,31 @@ class Bool {
     return 1;
   }
 
-  // TODO
-  static count(x: Bool | boolean[]): Field {
-    return new Field(0);
-  }
-
   static check(x: Bool): void {
     Snarky.field.assertBoolean(x.value);
   }
 
   static Unsafe = {
     /**
      * Converts a {@link Field} into a {@link Bool}. This is a **dangerous** operation
-     * as it assumes that the field element is either 1 or 0
-     * (which might not be true).
+     * as it assumes that the field element is either 0 or 1 (which might not be true).
+     *
+     * Only use this with constants or if you have already constrained the Field element to be 0 or 1.
+     *
      * @param x a {@link Field}
      */
-    ofField(x: Field | number | string | boolean): Bool {
-      if (typeof x === 'number') {
-        return new Bool(x === 1);
-      } else if (typeof x === 'string') {
-        return new Bool(x === '1');
-      } else if (typeof x === 'boolean') {
-        return new Bool(x);
-      } else {
-        return new Bool(x.value);
-      }
+    ofField(x: Field) {
```

Comment:
I refactored this function as I felt the additional options to pass in non-`Field`s were never used and made it less safe, and I added a runtime error